### PR TITLE
removed tap.sh and updated junit.py

### PR DESCRIPTION
**Pull request by rezgarshakeri**
As mentioned here https://github.com/CEED/libCEED/issues/1068 , the `tap.sh` is removed and `junit.py` is updated

### CODE REVIEW

File: tests/junit.py [hunk 1]
```diff
@@ -169,17 +178,40 @@ def run(test, backends):
...
+                    message = test_case.is_error() if isinstance(test_case.is_error(), str) else test_case.errors[0]['message']
+                    print('not ok {} - ERROR: {}'.format(test_index, message.strip()))
+                    print(test_case.errors[0]['output'].strip())
+                elif test_case.is_failure():
+                    message = test_case.is_failure() if isinstance(test_case.is_failure(), str) else test_case.failures[0]['message']
+                    print('not ok {} - FAIL: {}'.format(test_index, message.strip()))
+                    print(test_case.failures[0]['output'].strip())
+                elif test_case.is_skipped():
+                    message = test_case.is_skipped() if isinstance(test_case.is_skipped(), str) else test_case.skipped[0]['message']
+                    print('ok {} - SKIP: {}'.format(test_index, message.strip()))
```
Comment (jeremylt) [inline tests/junit.py:210]: I had to update my logic on handling these cases in Ratel due to a misunderstanding on my part: https://gitlab.com/micromorph/ratel/-/merge_requests/235
Note that the change included updating to a newer version of the Python package we're using here

Comment (rezgarshakeri) [reply tests/junit.py:210]: Should I add all the changes in that MR to this branch?

Comment (jeremylt) [reply tests/junit.py:210]: I'd recommend that. You can copy-paste the tests/junit-xml folder over from Ratel main and then that let's you use those changes from junit.py

Comment (rezgarshakeri) [reply tests/junit.py:210]: I copied from ratel. In this branch I get
```
make prove -j PROVE_OPTS=-v

All tests successful.
Files=208, Tests=1744, 28 wallclock secs ( 1.23 usr  0.36 sys + 19.61 cusr  5.44 csys = 26.64 CPU)
Result: PASS
```

But in main
we have more tests, why?
```
All tests successful.
Files=208, Tests=5232, 53 wallclock secs ( 3.27 usr  0.69 sys + 27.00 cusr 26.10 csys = 57.06 CPU)
Result: PASS
```

Comment (jeremylt) [reply tests/junit.py:210]: 5283/3 = 1744

Currently we report each test in triplicate, stdout, stderr, and the return code, but in Ratel we combined to a single report per test case

Comment (rezgarshakeri) [reply tests/junit.py:210]: Nice!

**Comment by jeremylt** [comment]
It looks like only python 2 is installed on the image that emulates different hardware?

File: tests/junit-xml/.travis.yml [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
 
 script:
-    - travis_retry tox
+  - travis_retry tox
+
+git:
+  depth: 1
 
 notifications:
   email:
```
Comment (jedbrown) [inline tests/junit-xml/.travis.yml:38]: We don't need or use this file. No idea if upstream does either.

File: tests/junit-xml/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+.PHONY : dist
+dist:
+	python setup.py sdist bdist_wheel
+
+.PHONY : clean
+clean:
+	find . -name "*.pyc" -delete
+	find . -name "__pycache__" -delete
+	rm -rf build
+	rm -rf dist
```
Comment (jedbrown) [inline tests/junit-xml/Makefile:17]: Don't need this file either.

**Review by jedbrown — APPROVED** [review]
Looks good, thanks. Please check `git grep tap.sh` for stale references.

**Comment by jeremylt** [comment]
The issue with the Actions job failing might be fixed by specifying the distro for the image used in addition to the arch in the yml for that job:

https://github.com/uraimo/run-on-arch-action

File: tests/junit-xml/.travis.yml (hunk 2 above)
Comment (rezgarshakeri) [reply tests/junit-xml/.travis.yml:38]: I deleted `travis.yml`

File: tests/junit-xml/Makefile (hunk 3 above)
Comment (rezgarshakeri) [reply tests/junit-xml/Makefile:17]: I deleted this Makefile as well.

**Comment by rezgarshakeri** [comment]
> 

How about here, should I delete `tap.sh`?
https://github.com/CEED/libCEED/blob/rezgar/tap-output/examples/mfem/Makefile#L51

**Comment by jeremylt** [comment]
> > 
> 
> 
> 
> How about here, should I delete `tap.sh`?
> 
> https://github.com/CEED/libCEED/blob/rezgar/tap-output/examples/mfem/Makefile#L51

I don't think it should have ever been there in the first place. I'd remove that rule

**Comment by rezgarshakeri** [comment]
> 

That's what we have
```
jobs:
  test:
    strategy:
      matrix:
        os: [ubuntu-20.04]
        compiler: [gcc-9]
        arch: [aarch64, ppc64le]
        distro: [ubuntu20.04]
```

**Comment by jeremylt** [comment]
> That's what we have
> 
> jobs:
>   test:
>     strategy:
>       matrix:
>         os: [ubuntu-20.04]
>         compiler: [gcc-9]
>         arch: [aarch64, ppc64le]
>         distro: [ubuntu20.04]

That's the distro for the parent container, not the container emulating the hardware.

**Comment by jeremylt** [comment]
You can strip out a bit more from the `junit-xml` folder: https://gitlab.com/micromorph/ratel/-/merge_requests/240

**Comment by rezgarshakeri** [comment]
> You can strip out a bit more from the `junit-xml` folder: https://gitlab.com/micromorph/ratel/-/merge_requests/240

Thanks!

**Comment by rezgarshakeri** [comment]
> > That's what we have
> > jobs:
> > test:
> > strategy:
> > matrix:
> > os: [ubuntu-20.04]
> > compiler: [gcc-9]
> > arch: [aarch64, ppc64le]
> > distro: [ubuntu20.04]
> 
> That's the distro for the parent container, not the container emulating the hardware.
> 
> I think that you need this:
> 
> ```diff
> diff --git a/.github/workflows/c-fortran-test-hardware.yml b/.github/workflows/c-fortran-test-hardware.yml
> index da7eed90..674e2521 100644
> --- a/.github/workflows/c-fortran-test-hardware.yml
> +++ b/.github/workflows/c-fortran-test-hardware.yml
> @@ -25,6 +25,7 @@ jobs:
>        id: runcmd
>        with:
>          arch: ${{ matrix.arch }}
> +        distro: ubuntu20.04
>          run: |
>            apt-get -y update
>            apt-get install -y build-essential
> ```

I'll add the `distro` to that yml file.

**Comment by jeremylt** [comment]
That action seems to have removed python3 from their Ubuntu20.04 image - that's rather rude

**Comment by jeremylt** [comment]
Unfortunately this is also needed because they intentionally stripped almost everything out of those images
```
diff --git a/.github/workflows/c-fortran-test-hardware.yml b/.github/workflows/c-fortran-test-hardware.yml
index da7eed90..89b09112 100644
--- a/.github/workflows/c-fortran-test-hardware.yml
+++ b/.github/workflows/c-fortran-test-hardware.yml
@@ -29,6 +29,7 @@ jobs:
           apt-get -y update
           apt-get install -y build-essential
           apt-get install -y gfortran
+          apt-get install -y python3
           uname -a
           make info
           make -j2
```

**Review by jeremylt — APPROVED** [review]

**Pull request merged**